internal/manifests/manifests.go

```diff
@@ -355,6 +361,29 @@ func GetClusterManifests(ctx context.Context, clusterID *strfmt.UUID, s3Client s
 	return manifestFiles, nil
 }
 
+// Return the list of the manifests provided by the user into a list of "{folder}/{filename}"
+func GetUserManifestSuffixes(ctx context.Context, clusterID *strfmt.UUID, s3Client s3wrapper.API) ([]string, error) {
```

Comment:
I am not sure I would call this `GetUserManifestSuffixes`, 

as far as I can see you are expecting an entry like 

```
filepath.Join(clusterId.String(), constants.ManifestMetadataFolder, "manifests", "first.yaml", constants.ManifestSourceUserSupplied)
``` 

to resolve to 

```
"manifests"/"first.yaml"
```

Seems that this is more about resolving the manifest names independent of any cluster id

Reply:
thanks, I think I will split this into 2 functions:
- GetManifestMetadata =>  returns full path of metadata
- ResolveManifestNamesFromMetadata => which takes full path of metadata and just extract folder and manifest names